libsolidity/parsing/Parser.cpp

```diff
@@ -1585,6 +1585,18 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 			nodeFactory.markEndPosition();
 		expression = nodeFactory.createNode<NewExpression>(typeName);
 	}
+	else if (m_scanner->currentToken() == Token::Payable)
+	{
+		expectToken(Token::Payable);
+		nodeFactory.markEndPosition();
+		auto expressionType = nodeFactory.createNode<ElementaryTypeName>(
+			ElementaryTypeNameToken(Token::Address, 160, 0),
+			boost::make_optional(StateMutability::Payable)
+		);
+		expression = nodeFactory.createNode<ElementaryTypeNameExpression>(expressionType);
+		if (m_scanner->currentToken() != Token::LParen)
+			parserError("Expected address to be converted to address payable.");
```

Comment:
What about "expected ("? I'm not sure I would understand it this way.

Reply:
Ok! Only "expected (" ?

Reply:
``expectToken(Token::LParen, false);`` should do it I think...

Reply:
But doesn't that also consume it?

Reply:
That's the ``false`` - the second argument is ``bool _advance``, defaulting to true.

Reply:
Hm it actually worked